assets/scripts/app/ScrollIndicators.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { registerKeypress } from './keypress'
+
+class ScrollIndicators extends React.Component {
+  static propTypes = {
+    posLeft: PropTypes.number.isRequired,
+    posRight: PropTypes.number.isRequired,
+    scrollStreet: PropTypes.func.isRequired,
+    scrollTop: PropTypes.number.isRequired
+  }
+
+  shouldComponentUpdate (nextProps) {
+    if (this.props.posLeft !== nextProps.posLeft &&
+        this.props.posRight !== nextProps.posRight) {
+      this.refs.left_indicator.innerHTML = Array(nextProps.posLeft + 1).join('‹')
+      this.refs.right_indicator.innerHTML = Array(nextProps.posRight + 1).join('›')
+      return true
+    }
+    // Checking if window was resized
+    if (this.props.scrollTop !== nextProps.scrollTop) {
+      this.updateIndicatorsPosition(nextProps.scrollTop)
+      return true
+    }
+    return false
+  }
+
+  componentDidMount () {
+    registerKeypress('left', (event) => { this.props.scrollStreet(true, event.shiftKey) })
+    registerKeypress('right', (event) => { this.props.scrollStreet(false, event.shiftKey) })
+  }
+
+  updateIndicatorsPosition = (scrollTop) => {
+    this.refs.left_indicator.style.top = scrollTop + 'px'
+    this.refs.right_indicator.style.top = scrollTop + 'px'
+  }
+
+  render () {
+    return (
+      <div className="street-scroll-indicators">
```

Comment:
Since both scroll indicators are in the same box, I think we can get away with setting the `top` style just once, on this element. Instead of having refs to the element, you can set the style directly in React: `<div style={{ top: scrollTop + 'px' }}>`. Since this gets updated on each `render()`, and render is only called because `PureComponent` does `shouldComponentUpdate` for us, we can remove the `updateIndicatorsPosition()` method.

Reply:
It's possible we have to refactor a bunch of CSS to make this work, but that might be a good idea anyway, to make the positioning a little easier to understand.